common/src/main/java/org/opensearch/sdk/DeleteCustomResponse.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+public class DeleteCustomResponse {
+    private final String id;
+    private final boolean deleted;
+
+    public DeleteCustomResponse(String id, boolean deleted) {
+        this.id = id;
+        this.deleted = deleted;
+    }
+
+    /**
+     * Returns the document id
+     * @return the id
+     */
+    public String id() {
+        return id;
+    }
+
+    /**
+     * Returns whether deletion was successful
+     * @return true if deletion was successful, false if the object was not found
```

Comment:
what happens if the object was found but some reason we couldn't delete.

Reply:
The onFailure / exception path will be followed instead.  This is the same behavior as the current DeleteResponse where you can have either DELETED (true) or NOT_FOUND (false) successful response.